pyface/undo/abstract_command.py

```diff
@@ -0,0 +1,76 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2007, Riverbank Computing Limited
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in enthought/LICENSE.txt and may be redistributed only
+# under the conditions described in the aforementioned license.  The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+# Thanks for using Enthought open source!
+#
+# Author: Riverbank Computing Limited
+# Description: <Enthought undo package component>
+# ------------------------------------------------------------------------------
+
+# Enthought library imports.
+from traits.api import Any, HasTraits, Str, provides
+
+# Local imports.
+from .i_command import ICommand
+
+
+@provides(ICommand)
+class AbstractCommand(HasTraits):
+    """The AbstractCommand class is an abstract base class that implements the
+    ICommand interface.
+    """
+
+    #### 'ICommand' interface #################################################
+
+    # This is the data on which the command operates.
+    data = Any
```

Comment:
I think it'll be good to follow the new convention of using `data = Any()` instead of `data = Any` for trait definitions. This applies here and all of the other trait definitions on new classes in this PR.